tests/queries/0_stateless/02494_query_cache_nested_query_bug.reference

```diff
@@ -1,2 +1,2 @@
-2
+1
```

Comment:
Why did the result change?

Reply:
To be honest: I don't know. The test is actually about testing the 2nd run of the same query, and for both cases (analyzer / non-analyzer), it returns 0 as the correct result.

But yes, the log output of of the Analyzer path looks more correct to me.

Non-analyzer log output: (contains `Aggregated.` twice)

```
/repo/ch4/build/programs (enable-02494_query_cache_nested_query_bug +=) $ chc --send_logs_level=trace --query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache=1, max_threads=1, allow_experimental_analyzer=0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.0" 2>&1
[ip-172-31-2-215] 2024.05.26 14:01:56.501740 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Debug> executeQuery: (from 127.0.0.1:40898) SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache=1, max_threads=1, allow_experimental_analyzer=0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.0 (stage: Complete)
[ip-172-31-2-215] 2024.05.26 14:01:56.502366 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> QueryCache: No query result found for query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache = 1, max_threads = 1, allow_experimental_analyzer = 0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability = 0."
[ip-172-31-2-215] 2024.05.26 14:01:56.504072 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
[ip-172-31-2-215] 2024.05.26 14:01:56.505646 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Debug> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Key condition: unknown
[ip-172-31-2-215] 2024.05.26 14:01:56.505744 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Filtering marks by primary and secondary keys
[ip-172-31-2-215] 2024.05.26 14:01:56.505880 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Debug> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Selected 2/2 parts by partition key, 2 parts by primary key, 2/2 marks by primary key, 2 marks to read from 2 ranges
[ip-172-31-2-215] 2024.05.26 14:01:56.505954 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Spreading mark ranges among streams (default reading)
[ip-172-31-2-215] 2024.05.26 14:01:56.506121 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Reading 1 ranges in order from part all_1_1_0, approx. 3 rows starting from 0
[ip-172-31-2-215] 2024.05.26 14:01:56.506224 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Reading 1 ranges in order from part all_2_2_0, approx. 3 rows starting from 0
[ip-172-31-2-215] 2024.05.26 14:01:56.507648 [ 2349375 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> AggregatingTransform: Aggregating
[ip-172-31-2-215] 2024.05.26 14:01:56.507783 [ 2349375 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> Aggregator: An entry for key=14612333762372916907 found in cache: sum_of_sizes=1, median_size=1
[ip-172-31-2-215] 2024.05.26 14:01:56.507860 [ 2349375 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> Aggregator: Aggregation method: without_key
[ip-172-31-2-215] 2024.05.26 14:01:56.508691 [ 2349375 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> AggregatingTransform: Aggregated. 6 to 1 rows (from 48.00 B) in 0.002128311 sec. (2819.137 rows/sec., 22.02 KiB/sec.)
[ip-172-31-2-215] 2024.05.26 14:01:56.508762 [ 2349375 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> Aggregator: Merging aggregated data
[ip-172-31-2-215] 2024.05.26 14:01:56.510339 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
[ip-172-31-2-215] 2024.05.26 14:01:56.513492 [ 2349397 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> AggregatingTransform: Aggregating
[ip-172-31-2-215] 2024.05.26 14:01:56.513597 [ 2349397 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> Aggregator: Aggregation method: without_key
[ip-172-31-2-215] 2024.05.26 14:01:56.513769 [ 2349397 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> AggregatingTransform: Aggregated. 2 to 1 rows (from 48.00 B) in 0.001372399 sec. (1457.302 rows/sec., 34.16 KiB/sec.)
[ip-172-31-2-215] 2024.05.26 14:01:56.513898 [ 2349397 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> Aggregator: Merging aggregated data
0.3333333333333333
[ip-172-31-2-215] 2024.05.26 14:01:56.514930 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Trace> QueryCache: Stored query result of query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache = 1, max_threads = 1, allow_experimental_analyzer = 0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability = 0."
[ip-172-31-2-215] 2024.05.26 14:01:56.515108 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Debug> executeQuery: Read 8 rows, 96.00 B in 0.014696 sec., 544.3658138268917 rows/sec., 6.38 KiB/sec.
[ip-172-31-2-215] 2024.05.26 14:01:56.515297 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Debug> MemoryTracker: Peak memory usage (for query): 341.36 KiB.
[ip-172-31-2-215] 2024.05.26 14:01:56.515364 [ 2348254 ] {aa2dbb12-7373-4420-8e54-51fe3bd2c0b5} <Debug> TCPHandler: Processed in 0.01595656 sec.

/repo/ch4/build/programs (enable-02494_query_cache_nested_query_bug +=) $ chc --send_logs_level=trace --query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache=1, max_threads=1, allow_experimental_analyzer=0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.0" 2>&1
[ip-172-31-2-215] 2024.05.26 14:01:59.224398 [ 2348254 ] {f940016f-e898-452a-bdeb-e8f448d01fd7} <Debug> executeQuery: (from 127.0.0.1:40912) SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache=1, max_threads=1, allow_experimental_analyzer=0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.0 (stage: Complete)
[ip-172-31-2-215] 2024.05.26 14:01:59.225012 [ 2348254 ] {f940016f-e898-452a-bdeb-e8f448d01fd7} <Trace> QueryCache: Query result found for query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache = 1, max_threads = 1, allow_experimental_analyzer = 0, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability = 0."
0.3333333333333333
[ip-172-31-2-215] 2024.05.26 14:01:59.226233 [ 2348254 ] {f940016f-e898-452a-bdeb-e8f448d01fd7} <Debug> executeQuery: Read 1 rows, 9.00 B in 0.003173 sec., 315.1591553734636 rows/sec., 2.77 KiB/sec.
[ip-172-31-2-215] 2024.05.26 14:01:59.226336 [ 2348254 ] {f940016f-e898-452a-bdeb-e8f448d01fd7} <Debug> TCPHandler: Processed in 0.004233717 sec.
```

Analyzer output: (contains `Aggregated.` once)

```
/repo/ch4/build/programs (enable-02494_query_cache_nested_query_bug +=) $ chc --send_logs_level=trace --query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache=1, max_threads=1, allow_experimental_analyzer=1, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.0" 2>&1
[ip-172-31-2-215] 2024.05.26 14:02:35.517660 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> executeQuery: (from 127.0.0.1:42938) SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache=1, max_threads=1, allow_experimental_analyzer=1, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability=0.0 (stage: Complete)
[ip-172-31-2-215] 2024.05.26 14:02:35.518225 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> QueryCache: No query result found for query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache = 1, max_threads = 1, allow_experimental_analyzer = 1, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability = 0."
[ip-172-31-2-215] 2024.05.26 14:02:35.520399 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Planner: Query to stage Complete
[ip-172-31-2-215] 2024.05.26 14:02:35.520833 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Planner: Query from stage FetchColumns to stage Complete
[ip-172-31-2-215] 2024.05.26 14:02:35.522732 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Key condition: unknown
[ip-172-31-2-215] 2024.05.26 14:02:35.522850 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Filtering marks by primary and secondary keys
[ip-172-31-2-215] 2024.05.26 14:02:35.522968 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Selected 2/2 parts by partition key, 2 parts by primary key, 2/2 marks by primary key, 2 marks to read from 2 ranges
[ip-172-31-2-215] 2024.05.26 14:02:35.523041 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Spreading mark ranges among streams (default reading)
[ip-172-31-2-215] 2024.05.26 14:02:35.523207 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Reading 1 ranges in order from part all_1_1_0, approx. 3 rows starting from 0
[ip-172-31-2-215] 2024.05.26 14:02:35.523310 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> default.tab (d5522565-e333-459c-9a73-00c043bc088e) (SelectExecutor): Reading 1 ranges in order from part all_2_2_0, approx. 3 rows starting from 0
[ip-172-31-2-215] 2024.05.26 14:02:35.524630 [ 2349369 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> AggregatingTransform: Aggregating
[ip-172-31-2-215] 2024.05.26 14:02:35.524755 [ 2349369 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Aggregator: Aggregation method: without_key
[ip-172-31-2-215] 2024.05.26 14:02:35.525246 [ 2349369 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> AggregatingTransform: Aggregated. 6 to 1 rows (from 48.00 B) in 0.001581775 sec. (3793.207 rows/sec., 29.63 KiB/sec.)
[ip-172-31-2-215] 2024.05.26 14:02:35.525320 [ 2349369 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Aggregator: Merging aggregated data
[ip-172-31-2-215] 2024.05.26 14:02:35.525396 [ 2349369 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Aggregator: Statistics updated for key=17741621837297420966: new sum_of_sizes=1, median_size=1
[ip-172-31-2-215] 2024.05.26 14:02:35.527152 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Planner: Query to stage Complete
[ip-172-31-2-215] 2024.05.26 14:02:35.527408 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Planner: Query to stage WithMergeableState only analyze
[ip-172-31-2-215] 2024.05.26 14:02:35.527666 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Planner: Query from stage FetchColumns to stage WithMergeableState only analyze
[ip-172-31-2-215] 2024.05.26 14:02:35.528260 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Planner: Query from stage WithMergeableState to stage Complete
[ip-172-31-2-215] 2024.05.26 14:02:35.530444 [ 2349385 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> MergingAggregatedTransform: Reading blocks of partially aggregated data.
[ip-172-31-2-215] 2024.05.26 14:02:35.530614 [ 2349385 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> MergingAggregatedTransform: Read 1 blocks of partially aggregated data, total 1 rows.
[ip-172-31-2-215] 2024.05.26 14:02:35.530694 [ 2349385 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Aggregator: Merging partially aggregated single-level data.
[ip-172-31-2-215] 2024.05.26 14:02:35.530777 [ 2349385 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Aggregator: Merged partially aggregated single-level data.
[ip-172-31-2-215] 2024.05.26 14:02:35.530864 [ 2349385 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> Aggregator: Converting aggregated data to blocks
[ip-172-31-2-215] 2024.05.26 14:02:35.531010 [ 2349385 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> Aggregator: Converted aggregated data to blocks. 1 rows, 8.00 B in 5.3049e-05 sec. (18850.497 rows/sec., 147.27 KiB/sec.)
0.3333333333333333
[ip-172-31-2-215] 2024.05.26 14:02:35.531924 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Trace> QueryCache: Stored query result of query "SELECT count(a) / (SELECT sum(a) FROM tab) FROM tab SETTINGS use_query_cache = 1, max_threads = 1, allow_experimental_analyzer = 1, merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability = 0."
[ip-172-31-2-215] 2024.05.26 14:02:35.532097 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> executeQuery: Read 7 rows, 64.00 B in 0.015778 sec., 443.6557231588287 rows/sec., 3.96 KiB/sec.
[ip-172-31-2-215] 2024.05.26 14:02:35.532279 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> MemoryTracker: Peak memory usage (for query): 394.34 KiB.
[ip-172-31-2-215] 2024.05.26 14:02:35.532346 [ 2348254 ] {ab265876-ee22-4f37-a378-75aa5692c955} <Debug> TCPHandler: Processed in 0.016958615 sec.
```